docs/reference/rest-api/security/bulk-create-roles.asciidoc

```diff
@@ -0,0 +1,342 @@
+                            "title",
+                            "body"
+                        ]
+                    },
+                    "query": "{\"match\": {\"title\": \"foo\"}}"
+                }
+            ],
+            "applications": [
+                {
+                    "application": "myapp",
+                    "privileges": [
+                        "admin",
+                        "read"
+                    ],
+                    "resources": [
+                        "*"
+                    ]
+                }
+            ],
+            "run_as": [
+                "other_user"
+            ],
+            "metadata": {
+                "version": 1
+            }
+        },
+        "my_user_role": {
+            "cluster": [
+                "all"
+            ],
+            "indices": [
+                {
+                    "names": [
+                        "index1"
+                    ],
+                    "privileges": [
+                        "read"
+                    ],
+                    "field_security": {
+                        "grant": [
+                            "title",
+                            "body"
+                        ]
+                    },
+                    "query": "{\"match\": {\"title\": \"foo\"}}"
+                }
+            ],
+            "applications": [
+                {
+                    "application": "myapp",
+                    "privileges": [
+                        "admin",
+                        "read"
+                    ],
+                    "resources": [
+                        "*"
+                    ]
+                }
+            ],
+            "run_as": [
+                "other_user"
+            ],
+            "metadata": {
+                "version": 1
+            }
+        }
+    }
+}
+--------------------------------------------------
+
+A successful call returns a JSON structure that shows whether the roles has been
+created, updated or was a noop in the same order as in the bulk request.
+
+[source,console-result]
+--------------------------------------------------
+{
+    "took": 45,
+    "errors": false,
+    "items": [
+        {
+            "result": "created", <1>
+            "status": 201, <2>
+            "_id": "my_admin_role"
+        },
+        {
+            "result": "created",
+            "status": 201,
+            "_id": "my_user_role"
+        }
+    ]
+}
+--------------------------------------------------
+// TESTRESPONSE[s/"took": 45/"took": $body.took/]
+
+<1> When an existing role is updated, `result` is set to `updated` or when no changes were made it's set to `noop`.
+<2> When an existing role is updated, `status` is set to `200`.
+
+Errors are handled individually for each role create or update and the API therefore allows partial success.
+
+The following query would throw an error for `my_admin_role` but would be successful for the `my_user_role`.
```

Comment:
why would it throw an error? if it's malformed we should say that, but if it's due to the context of the cluster then we should explain more about the hypothetical situation

Reply:
It would fail because the privilege doesn't exist. I'll add some context about that. Thanks for pointing it out!